### PR TITLE
Custom MI related exception

### PR DESCRIPTION
**Pull request by pallar-ms**
Changes include:
* Add new custom exceptions to summarize managed identity related errors
  - if the custom auth type is used and the credential bundle for the managed identity is not present in the key vault
  - if the underlying mi is no longer present
  - use localized strings for the exception message

### CODE REVIEW

File: src/lib/Microsoft.Health.Events/Resources/EventResources.resx [hunk 1]
```diff
@@ -0,0 +1,144 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (rogordon01) [inline src/lib/Microsoft.Health.Events/Resources/EventResources.resx:1]: Is this file needed?

File: src/lib/Microsoft.Health.Events/Resources/EventResources.Designer.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
+﻿//------------------------------------------------------------------------------
```
Comment (rogordon01) [inline src/lib/Microsoft.Health.Events/Resources/EventResources.Designer.cs:1]: Is this file needed?

File: src/lib/Microsoft.Health.Events/Microsoft.Health.Events.csproj [hunk 3]
```diff
@@ -39,4 +39,17 @@
     <ProjectReference Include="..\Microsoft.Health.Common\Microsoft.Health.Common.csproj" />
     <ProjectReference Include="..\Microsoft.Health.Logger\Microsoft.Health.Logging.csproj" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (rogordon01) [inline src/lib/Microsoft.Health.Events/Microsoft.Health.Events.csproj:42]: Not sure if we needed these files. If not you can revert this change

File: src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs [hunk 4]
```diff
@@ -68,24 +72,40 @@ public static (Exception customException, string errorName) CustomizeException(E
...
+                    var msalErrorCode = ((MsalServiceException)exception).ErrorCode;
+                    message = EventResources.ManagedIdentityAuthenticationError;
+                    return (new ManagedIdentityAuthenticationError(message, exception, msalErrorCode), msalErrorCode);
+
+                case RequestFailedException _:
+                    var errorCode = ((RequestFailedException)exception).ErrorCode;
+
+                    if (string.Equals(errorCode, EventResources.SecretNotFound, StringComparison.CurrentCultureIgnoreCase) || message.Contains(EventResources.SecretNotFound, StringComparison.CurrentCultureIgnoreCase))
+                    {
+                        message = EventResources.ManagedIdentityCredentialNotFound;
```
Comment (rogordon01) [inline src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs:89]: If we get here is it always guaranteed that we have an issue with Managed Identity? Looking at the code I think that we also support [ConnectionString](https://microsofthealth.visualstudio.com/Health/_git/workspace-platform?path=/iot-connector/core/iomt-fhir/Microsoft.Health.Fhir.Ingest.Console.PaaS/EventHubs/EventHubConsumerClientFactory.cs&version=GBmain&line=32&lineEnd=33&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).

File: src/lib/Microsoft.Health.Events/Resources/EventResources.resx (hunk 1 above)
Comment (pallar-ms) [reply src/lib/Microsoft.Health.Events/Resources/EventResources.resx:1]: Yes, I added this resource file to localize the strings that we expose to the customers, in this case I've added the custom exception messages.

File: src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs (hunk 4 above)
Comment (pallar-ms) [reply src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs:89]: If using the ConnectionString auth type, since there is no secret involved that we fetch from the key vault, I think only in the custom auth type, we use the credential bundle and can expect to get this error if it doesn't exist in the keyvault.

**Review by rogordon01 — APPROVED** [review]

**Review by namalu — APPROVED** [review]

File: src/lib/Microsoft.Health.Events/Microsoft.Health.Events.csproj (hunk 3 above)
Comment (pallar-ms) [reply src/lib/Microsoft.Health.Events/Microsoft.Health.Events.csproj:42]: Yes, I added this resource file to localize the strings that we expose to the customers, in this case I've added the custom exception messages.
